### ISSUE TITLE
Availability of the link feature

### ISSUE THREAD

**Issue by jchristgit** [basho/riak#968]
We are currently using links for establishing a hierarchy between Riak entries (main objects and sub objects, with one level of nesting), but it seems like links were deprecated in 2.0. The [protocol buffer documentation](https://docs.riak.com/riak/kv/2.0.0/developing/api/protocol-buffers/fetch-object.1.html) reads:

> links — This parameter is associated with the now-deprecated link walking feature and should not be used by Riak clients

Will link metadata be part of the upcoming release or is there a way to opt into it? Alternatively, which alternatives could be used instead of this links?

**Comment by martinsumner** [comment]
Apologies for the long delay in picking up on the question.

Although the link feature was deprecated, I don't believe it has been removed.  I've just verified that the [test of this feature](https://github.com/basho/riak_test/blob/develop-2.9/tests/verify_link_walk_urls.erl) still passes on the latest version.  Perhaps

There's no intention to to take active action to remove it from releases going forward that I'm aware of.  However, it is not something we have an active focus on testing in the release cycle, as the main Riak users involved in development don't use links.  

There may well be hidden and unexpected issues when combining links with other features, which may have been the motivation behind the original deprecation warning.  Although there may be no intention to stop links form working in the future, they may stop working.

In terms of alternatives, I think in a lot of cases link-walking managed by Riak isn't seen as necessarily a big efficiency.  So when Riak users have object relationships they tend to encode those relationships into the body or header of the application object, and then walk links through their application themselves.
